### PR TITLE
Fix case where number of shuffle writer threads is set to 0

### PR DESCRIPTION
**Pull request by abellina**
Signed-off-by: Alessandro Bellina <abellina@nvidia.com>

Closes: https://github.com/NVIDIA/spark-rapids/issues/6621.

This fixes a bug when `spark.rapids.shuffle.multiThreaded.writer.threads=0`, which is documented to be supported and means we fallback the shuffle writer to the one supplied by Spark without any modifications by us. Before this change, there could be a divide-by-zero error when we try to submit writer tasks.

### CODE REVIEW

**Comment by abellina** [comment]
build

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManagerBase.scala [hunk 1]
```diff
@@ -1181,10 +1181,16 @@ class RapidsShuffleInternalManagerBase(conf: SparkConf, val isDriver: Boolean)
           gpu.dependency.metrics)
       case bmssh: BypassMergeSortShuffleHandle[_, _] =>
         bmssh.dependency match {
-          case gpuDep: GpuShuffleDependency[_, _, _] if gpuDep.useMultiThreadedShuffle =>
-            // cast the handle with specific generic types due to type-erasure
+          case gpuDep: GpuShuffleDependency[_, _, _]
+              if gpuDep.useMultiThreadedShuffle &&
+                  rapidsConf.shuffleMultiThreadedWriterThreads > 0 =>
+            // use the threaded writer if the number of threads specified is 1 or above,
+            // with 0 threads we fallback to the Spark-provided writer.
```
Comment (tgravescs) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManagerBase.scala:1188]: is it obvious in the logs that we fallback? If not do we want to add a log message here?

Comment (abellina) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManagerBase.scala:1191]: I added more info in the log line for the shuffle manager, hope that helps address. https://github.com/NVIDIA/spark-rapids/pull/6641/commits/e7ce535a20615074bd32b46dc0b8303ded3c04d7. We document the fallback in the docs:

spark.rapids.shuffle.multiThreaded.writer.threads | The number of threads to use for writing shuffle blocks per executor in the RAPIDS shuffle manager configured in "MULTITHREADED" mode. There are two special values: 0 = feature is disabled, falls back to Spark built-in shuffle writer; 1 = our implementation of Spark's built-in shuffle writer with extra metrics.
-- | --

**Comment by abellina** [comment]
build

**Pull request merged**